### ISSUE TITLE
CP2K+ELPA-2018.11.001 possibly broken

### ISSUE THREAD

**Issue by dev-zero** [cp2k/cp2k#190]
Testing CP2K (naively) with ELPA-2018.11.001 results in completely different (and very likely wrong) values and/or IEEE_UNDERFLOW errors:

Example:

```
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/data/tiziano/cp2k/regtesting/local/psmp/TEST-local-psmp-2019-01-31_17-20-53/QS/regtest-elpa-qr/si128-elpa-qr-unsafe.inp.out
 PW_GRID|   Bounds   1             -2       1                Points:           4
 PW_GRID|   Bounds   2             -4       3                Points:           8
 PW_GRID|   Bounds   3             -4       3                Points:           8
 PW_GRID| Volume element (a.u.^3)   16.87         Volume (a.u.^3)      4318.3458
 PW_GRID| Grid span                                                    FULLSPACE
 PW_GRID|   Distribution                         Average         Max         Min
 PW_GRID|   G-Vectors                              128.0         132         124
 PW_GRID|   G-Rays                                  32.0          33          31
 PW_GRID|   Real Space Points                      128.0         128         128

 POISSON| Solver                                                        PERIODIC
 POISSON| Periodicity                                                        XYZ

 RS_GRID| Information for grid number                                          1
 RS_GRID|   Bounds   1            -12      11                Points:          24
 RS_GRID|   Bounds   2            -22      22                Points:          45
 RS_GRID|   Bounds   3            -22      22                Points:          45
 RS_GRID| Real space fully replicated
 RS_GRID| Group size                                                           1

 RS_GRID| Information for grid number                                          2
 RS_GRID|   Bounds   1             -6       5                Points:          12
 RS_GRID|   Bounds   2            -12      11                Points:          24
 RS_GRID|   Bounds   3            -12      11                Points:          24
 RS_GRID| Real space fully replicated
 RS_GRID| Group size                                                           1

 RS_GRID| Information for grid number                                          3
 RS_GRID|   Bounds   1             -4       3                Points:           8
 RS_GRID|   Bounds   2             -7       7                Points:          15
 RS_GRID|   Bounds   3             -7       7                Points:          15
 RS_GRID| Real space fully replicated
 RS_GRID| Group size                                                           1

 RS_GRID| Information for grid number                                          4
 RS_GRID|   Bounds   1             -2       1                Points:           4
 RS_GRID|   Bounds   2             -4       3                Points:           8
 RS_GRID|   Bounds   3             -4       3                Points:           8
 RS_GRID| Real space fully replicated
 RS_GRID| Group size                                                           1

 Number of electrons:                                                        128
 Number of occupied orbitals:                                                 64
 Number of molecular orbitals:                                                84

 Number of orbital functions:                                                128
 Number of independent orbital functions:                                    128

 Extrapolation method: initial_guess


 SCF WAVEFUNCTION OPTIMIZATION

  Step     Update method      Time    Convergence         Total energy    Change
  ------------------------------------------------------------------------------

ELPA| Matrix diagonalization information
ELPA| Matrix order             :            128
ELPA| Matrix block size        :             32
ELPA| Kernel                   :        GENERIC
ELPA| QR step requested        :              T
ELPA| Use potentially unsafe QR:              T
ELPA| Matrix is suitable for QR:              T
ELPA| Matrix block size check was bypassed
 Kernel time:     0.001 MFlops:  8.08709E+03
 Kernel time:     0.001 MFlops:  7.96238E+03
     1 NoMix/Diag. 0.60E+00    0.4     2.00000000      -128.5044566878 -1.29E+02

ELPA| Matrix diagonalization information
ELPA| Matrix order             :            128
ELPA| Matrix block size        :             32
ELPA| Kernel                   :        GENERIC
ELPA| QR step requested        :              T
ELPA| Use potentially unsafe QR:              T
ELPA| Matrix is suitable for QR:              T
ELPA| Matrix block size check was bypassed
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG
--------------------------------------------------------------------------
mpiexec has exited due to process rank 0 with PID 0 on
node tcopt12 exiting improperly. There are three reasons this could occur:

1. this process did not call "init" before exiting, but others in
the job did. This can cause a job to hang indefinitely while it waits
for all processes to call "init". By rule, if one process calls "init",
then ALL processes must call "init" prior to termination.

2. this process called "init", but exited without calling "finalize".
By rule, all processes that call "init" MUST call "finalize" prior to
exiting or it will be considered an "abnormal termination"

3. this process called "MPI_Abort" or "orte_abort" and the mca parameter
orte_create_session_dirs is set to false. In this case, the run-time cannot
detect that the abort call was an abnormal termination. Hence, the only
error message you will receive is this one.

This may have caused other processes in the application to be
terminated by signals sent by mpiexec (as reported here).

You can avoid this message by specifying -quiet on the mpiexec command line.
--------------------------------------------------------------------------
EXIT CODE:  1  MEANING:  RUNTIME FAIL
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
/data/tiziano/cp2k/regtesting/local/psmp/TEST-local-psmp-2019-01-31_17-20-53/QS/regtest-elpa/H2O-6.inp.out : 
 Total energy: : ref = -17.14603641576940 new = -0.92894878465317  
 relative error :   1.74574615e+01 >  numerical tolerance = 2e-14  
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

Using the following 
[enable_elpa-2018.11.001.patch.txt](https://github.com/cp2k/cp2k/files/2818178/enable_elpa-2018.11.001.patch.txt) to enable ELPA-2018.11.001 support.

**Comment by oschuett** [comment]
Interestingly, the problem was [not observed](https://github.com/cp2k/cp2k/pull/396#issuecomment-497264750) with Intel 19.0.3 while with Intel 18.0.4. it doesn't even compile.

**Comment by mkrack** [comment]
ELPA 2018.11.001 works also with gfortran 6.4.0 and mpich/3.2.1. I note, however, that the ELPA 2018.11.001 library has been built with AVX disabled, since our local cluster includes still some old nodes which provide only SSE2.

**Comment by mkrack** [comment]
I observe the same behaviour as @dev-zero now for the psmp version which could imply that the issue is related to OpenMP, since the popt version was fine. @dev-zero did you test also the popt version?

**Comment by mkrack** [comment]
Using the ELPA 2018.11.001 library compiled without OpenMP support for building cp2k.psmp seems to be a viable workaround.

**Comment by dev-zero** [comment]
With the adaption to the new ELPA API in PR #477 I can't reproduce this anymore. I guess I was seeing the same thing as @mkrack

**Comment by dev-zero** [comment]
closing since not reproducible with new ELPA API and I just merged #477
